### ISSUE TITLE
[PRO] When sorting by array field (PostgreSQL) it raises an exception graphql/pro/relation_connection/condition.rb:35

### ISSUE THREAD

**Issue by krapans** [rmosolgo/graphql-ruby#3166]
**Describe the bug**

Currently there's no support for sorting by array fields which are widely used within PostgreSQL databases. Let's take this query as an example where value in this case is our sorting field.

```sql
SELECT id, ARRAY[users.first_name, users.last_name] as value FROM users ORDER BY value ASC
```

This would raise an error something like this.

```ruby
ActiveRecord::StatementInvalid:
    PG::InvalidTextRepresentation: ERROR:  malformed array literal: "John"
    
    DETAIL:  Array value must start with "{" or dimension information.
```

The reason why this happens, is because `graphql/pro/relation_connection/condition.rb:35` expects always one type of a cursor and when we receive array, it's not enough just to parse it as a JSON.

```ruby
# graphql/pro/relation_connection/condition.rb:35

cursor_values = JSON.parse(cursor)
# When sorting by array field, value of cursor_values is a bit different
p  cursor_values # [["John", "Doe"], 1]
```

This particular array format is not really the format that PostgreSQL supports, thus we get an error. Current solution that we implemented locally for our case is patching a particular line of code.

```ruby
# graphql/pro/relation_connection/condition.rb:35

cursor_values = JSON.parse(cursor).map do |value|
    value.is_a?(Array) ? PG::TextEncoder::Array.new.encode(value) : value
end
# Now sorting value is encoded in correct format for performing sort by array field
p cursor_values # ["{John,Doe}", 1]
```

**Versions**

`graphql`: 1.11.4
`graphql-pro`: 1.15.4
`graphql-batch`: 0.4.3
`rails`: 6.0.3.2

**Steps to reproduce**

Create a sorting by custom `ARRAY[]` structure SQL query and use that particular field as a `sort_by` field.

**Expected behavior**

Ability to sort by array fields in PostgreSQL with Graphql API.

**Actual behavior**

Raises an exception

```ruby
ActiveRecord::StatementInvalid:
    PG::InvalidTextRepresentation: ERROR:  malformed array literal: "John"
    
    DETAIL:  Array value must start with "{" or dimension information.
```

**Comment by rmosolgo** [comment]
Sorry about the hangup, and thanks for the detailed report! I'll take a look and add some support for this.

**Comment by rmosolgo** [comment]
I just shipped graphql-pro 1.15.7 which should handle Postgres arrays properly, please let me know if you run into any more trouble. Thanks again for reporting it!
